chat-plugins/prowrestling.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Professional Wrestling Room Plugin
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ *
+ * This handles submitting questions for the podcast that the room runs periodically,
+ * instead of having to use another source to ask questions.
+ *
+ * @license MIT license
+ */
+
+'use strict';
+
+const FS = require('../fs');
+const PW_DATA_PATH = 'config/chat-plugins/prowrestling-questions.json';
+
+const MAX_QUESTION_LENGTH = 500;
+
+let pwData = {};
+try {
+	let pwData = require(`..${PW_DATA_PATH}`);
+} catch (e) {} // file doesn't exist (yet)
+
+class ProWrestling {
+	constructor() {
+		this.loadQuestions();
+	}
+	loadQuestions() {
+		FS(PW_DATA_PATH).readTextIfExists().then(data => {
+			if (!data) return;
+			pwData = JSON.parse(data);
+		});
+	}
+	saveQuestions() {
+		FS(PW_DATA_PATH).writeSync(JSON.stringify(pwData));
```

Comment:
This can deadlock. Don't allow multiple calls to the method to be able to try to write to the same file at the same time